pkg/agent/manager/storage.go

```diff
@@ -28,17 +28,17 @@ func ReadBundle(bundleCachePath string) ([]*x509.Certificate, error) {
 	return bundle, nil
 }
 
-// storeBundle writes manager's bundle to disk into bundleCachePath. Returns nil if all went
+// StoreBundle writes the bundle to disk into bundleCachePath. Returns nil if all went
 // fine, otherwise ir returns an error.
-func (m *manager) storeBundle() error {
+func StoreBundle(bundleCachePath string, bundle []*x509.Certificate) error {
 	// Write all certs to data bytes buffer.
 	data := &bytes.Buffer{}
-	for _, cert := range m.cache.Bundle() {
+	for _, cert := range bundle {
 		data.Write(cert.Raw)
 	}
 
 	// Write data to disk.
-	return ioutil.WriteFile(m.bundleCachePath, data.Bytes(), 0600)
+	return ioutil.WriteFile(bundleCachePath, data.Bytes(), 0600)
```

Comment:
at some point we might want to consider the writing to a temp file and then renaming over the top.

Reply:
That's a good idea, maybe we could create an issue to track it.